app/controllers/users_controller.rb

```diff
@@ -7,6 +7,7 @@ class UsersController < ApplicationController
   skip_before_action :require_login, :authorize, :session_expiry, :update_activity_time, :set_taxonomy, :set_gettext_locale_db, :only => [:login, :logout, :extlogout]
   skip_before_action :authorize, :only => :extlogin
   after_action       :update_activity_time, :only => :login
+  before_action      :verify_active_session, only: :login, unless: -> { request.post? }
```

Comment:
I'd suggest sticking with the same hash style as the previous line for readability.